### PR TITLE
More onboarding doc updates

### PR DESCRIPTION
**Pull request by lintool**

### CODE REVIEW

File: docs/start-here.md [hunk 1]
```diff
@@ -23,7 +23,8 @@ This is the definition I typically give:
 > Given an information need expressed as a query _q_, the text ranking task is to return a ranked list of _k_ texts {_d<sub>1</sub>_, _d<sub>2</sub>_ ... _d<sub>k</sub>_} from an arbitrarily large but finite collection
 of texts _C_ = {_d<sub>i</sub>_} that maximizes a metric of interest, for example, nDCG, AP, etc.
 
-This problem has been given various names, e.g., the search problem, the information retrieval problem, the text ranking problem, etc.
+This problem has been given various names, e.g., the search problem, the information retrieval problem, the text ranking problem, the top-_k_ document retrieval problem, etc.
+In most contexts, "ranking" and "retrieval" is used interchangeably.
```
Comment (sahel-sh) [inline docs/start-here.md:27]:
```suggestion
In most contexts, "ranking" and "retrieval" are used interchangeably.
```

File: docs/start-here.md [hunk 2]
```diff
@@ -268,5 +269,7 @@ By now you should be able to connect the concepts we introduced to how they mani
 From here, you're now ready to proceed to try and reproduce the [BM25 Baselines for MS MARCO Passage Ranking
 ](experiments-msmarco-passage.md).
 
+Before you move on, however, add an entry in the "Reproduction Log" at the bottom of this page, following the same format: use `yyyy-mm-dd`, make sure you're using a commit id that's on the main trunk of Anserini, and use a 7-hexadecimal prefix for the link anchor text.
```
Comment (sahel-sh) [inline docs/start-here.md:272]:
```suggestion
Before you move on, however, add an entry in the "Reproduction Log" at the bottom of this page, following the same format: use `yyyy-mm-dd`, make sure you're using a commit id that's on the main trunk of Anserini, and use its 7-hexadecimal prefix for the link anchor text.
```

File: docs/experiments-msmarco-passage.md [hunk 3]
```diff
@@ -280,6 +292,8 @@ In short, it's complicated.
 At this time, look back through the learning outcomes again and make sure you're good.
 As a next step in the onboarding path, you basically [do the same thing again in Python with Pyserini](https://github.com/castorini/pyserini/blob/master/docs/experiments-msmarco-passage.md) (as opposed to Java with Anserini here).
 
+Before you move on, however, add an entry in the "Reproduction Log" at the bottom of this page, following the same format: use `yyyy-mm-dd`, make sure you're using a commit id that's on the main trunk of Anserini, and use a 7-hexadecimal prefix for the link anchor text.
```
Comment (sahel-sh) [inline docs/experiments-msmarco-passage.md:295]:
```suggestion
Before you move on, however, add an entry in the "Reproduction Log" at the bottom of this page, following the same format: use `yyyy-mm-dd`, make sure you're using a commit id that's on the main trunk of Anserini, and use its 7-hexadecimal prefix for the link anchor text.
```

**Comment by lintool** [comment]
Also, I think "text retrieval task" sounds better in the definition, vs. "text ranking".

**Review by sahel-sh — APPROVED** [review]

**Pull request merged**